engine/verification/verifier/verifiers.go

```diff
@@ -0,0 +1,207 @@
+	}
+
+	from := lastSealed.Height - k + 1
+
+	// root block is not verifiable, because it's sealed already.
+	// the first verifiable is the next block of the root block
+	firstVerifiable := root + 1
+
+	if from < firstVerifiable {
+		from = firstVerifiable
+	}
+	to := lastSealed.Height
+
+	log.Info().Msgf("verifying blocks from %d to %d", from, to)
+
+	for height := from; height <= to; height++ {
+		log.Info().Uint64("height", height).Msg("verifying height")
+		err := verifyHeight(height, storages.Headers, chunkDataPacks, storages.Results, state, verifier)
+		if err != nil {
+			return fmt.Errorf("could not verify height %d: %w", height, err)
+		}
+	}
+
+	return nil
+}
+
+// VerifyRange verifies all chunks in the results of the blocks in the given range.
+func VerifyRange(
+	from, to uint64,
+	chainID flow.ChainID,
+	protocolDataDir string, chunkDataPackDir string,
+) error {
+	closer, storages, chunkDataPacks, state, verifier, err := initStorages(chainID, protocolDataDir, chunkDataPackDir)
+	if err != nil {
+		return fmt.Errorf("could not init storages: %w", err)
+	}
+	defer func() {
+		err := closer()
+		if err != nil {
+			log.Error().Err(err).Msg("failed to close storages")
+		}
+	}()
+
+	log.Info().Msgf("verifying blocks from %d to %d", from, to)
+
+	root := state.Params().SealedRoot().Height
+
+	if from <= root {
+		return fmt.Errorf("cannot verify blocks before the root block, from: %d, root: %d", from, root)
+	}
+
+	for height := from; height <= to; height++ {
+		log.Info().Uint64("height", height).Msg("verifying height")
+		err := verifyHeight(height, storages.Headers, chunkDataPacks, storages.Results, state, verifier)
+		if err != nil {
+			return fmt.Errorf("could not verify height %d: %w", height, err)
+		}
+	}
+
+	return nil
+}
+
+func initStorages(chainID flow.ChainID, dataDir string, chunkDataPackDir string) (
+	func() error,
+	*storage.All,
+	storage.ChunkDataPacks,
+	protocol.State,
+	module.ChunkVerifier,
+	error,
+) {
+	db := common.InitStorage(dataDir)
+
+	storages := common.InitStorages(db)
+	state, err := common.InitProtocolState(db, storages)
+	if err != nil {
+		return nil, nil, nil, nil, nil, fmt.Errorf("could not init protocol state: %w", err)
+	}
+
+	chunkDataPackDB, err := storagepebble.OpenDefaultPebbleDB(chunkDataPackDir)
+	if err != nil {
+		return nil, nil, nil, nil, nil, fmt.Errorf("could not open chunk data pack DB: %w", err)
+	}
+	chunkDataPacks := storagepebble.NewChunkDataPacks(metrics.NewNoopCollector(),
+		chunkDataPackDB, storages.Collections, 1000)
+
+	verifier := makeVerifier(log.Logger, chainID, storages.Headers)
+	closer := func() error {
+		err := db.Close()
+		if err != nil {
+			return fmt.Errorf("failed to close protocol db: %w", err)
+		}
+		err = chunkDataPackDB.Close()
+		if err != nil {
+			return fmt.Errorf("failed to close chunk data pack db: %w", err)
+		}
+		return nil
+	}
```

Comment:
Maybe we should try to close `chunkDataPackDB` even when `db.Close()` returns error.

```suggestion
	closer := func() error {
	        var dbErr, chunkDataPackDBErr error
		
		if err := db.Close(); err != nil {
			dbErr = fmt.Errorf("failed to close protocol db: %w", err)
		}
		
		if err := chunkDataPackDB.Close(); err != nil {
			chunkDataPackDBErr = fmt.Errorf("failed to close chunk data pack db: %w", err)
		}
		return errors.Join(dbErr, chunkDataPackDBErr)
	}
```